### DISCUSSION TITLE
The version of libwebp used by ImageSharp is not recognized by the BDBA

### DISCUSSION THREAD

**Discussion by manupstairs** [SixLabors/ImageSharp#2410] [Q&A] [answered]
Hi, I have a project that indirectly references **ImageSharp**. 
When I use **BDBA**(Black Duck Binary Analysis) to scan project, then I find **BDBA** can't reconginzed the version of **libwebp** (**ImageSharp** use it).
For components of unknown versions, the **BDBA** displays all historical vulnerabilities.
<img width="652" alt="image" src="https://user-images.githubusercontent.com/1835049/227129723-3580e663-e5d7-486e-b4b8-4955344cc994.png">

My question is that, is there any way to confirm **libwebp** verison in **ImageSharp**? Any document, link, comment can indicate the version?
Thank you very much!

**Comment by tocsoft** [comment, marked as answer]
ImageSharp does not use libwebp... imagesharp is completly managed code so we have our own custom implementations of webp encodes/decoders.

**Comment by manupstairs** [comment]
Thanks @tocsoft !
That means this is a false positive report by BDBA.
Since imagesharp has own custom implementations of webp encodes/decoders. 
So I think there is no vulnerability just like above image second line show.

**Comment by Myster** [comment]
Does this mean the CVE-2023-4863 vulnerability or similar does not exist?
Is it possible when translating the original into managed code that the vulnerability was included?
Since the vulnerability relates to buffer overflow I assume this would only be possible if 'unsafe' keyword was used in the c# version.

**Reply by JimBobSquarePants** [reply]
There's no unsafe code in that area so the worst thing that would happen is an argument out of range exception. 

However.... I would like to see the diff from the relevant patch applied so that was avoided.

https://github.com/webmproject/libwebp/commit/902bc9190331343b2017211debcec8d2ab87e17a#diff-048cd0744572f0f96c1fd0a2115c8f6b4019473409d3584a348fce0f9e522b6c
